sympy/concrete/summations.py

```diff
@@ -1535,10 +1545,15 @@ def get_residue_factor(numer, denom, alternating):
         residues = [residue(residue_factor, z, root) for root in nonint_roots]
         return -S.Pi * sum(residues)
 
-    if not (a.is_finite and b is S.Infinity):
-        return None
+    if not ('even' == get_function_symmetery(numer, denom)):
+        #for odd function flip the limit and negate
+        #limit (-oo, a) is flipped to (-a, oo) and the answer is negated
+        if ('odd' == get_function_symmetery(numer, denom)):
```

Comment:
The `get_function_symmetry` function should not be called more than once. Also the word symmetry is mispelt.